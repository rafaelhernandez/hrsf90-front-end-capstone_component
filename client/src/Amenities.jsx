import React from 'react';
import AmenityItem from './AmenityItem.jsx';

const Amenities = (props) => (
  <div className="div-description-amenities">
    {props.amenities.map((amenity, idx) => <AmenityItem key={idx} amenity={amenity}></AmenityItem>)}
  </div>
);
export default Amenities;
