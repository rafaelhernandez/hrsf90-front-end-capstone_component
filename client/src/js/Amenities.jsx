import React from 'react';
import AmenityItem from './AmenityItem.jsx';
import ModalAmenities from './ModalAmenities.jsx';
import SVGIcons from './SVGIcons.js';
import ReactModal from 'react-modal';

class Amenities extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false
    };
    this.amenities = [[],[]];
    this.handleOpenModal = this.handleOpenModal.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
  }

  componentDidMount() {
    let num = 0;
    for (let i = 0; i < this.props.amenities.length; ++i) {
      if (this.props.amenities[i] in SVGIcons) {
        this.amenities[num % 2].push(this.props.amenities[i]);
        ++num;
      }
      if (num > 5) {
        break;
      }
    }
  }

  componentWillMount() {
    ReactModal.setAppElement('body');
  }

  handleOpenModal () {
    this.setState({ showModal: true });
  }

  handleCloseModal () {
    this.setState({ showModal: false });
  }

  render() {
    return (
      <div className="div-description-amenities">
        <div className="div-description-section-header">
          Amenities
        </div>
        <div className="div-description-amenities-items">
          <div className="div-description-amenities-left">
            {this.amenities[0].map((amenity, idx) => <AmenityItem key={idx} amenity={amenity}></AmenityItem>)}
          </div>
          <div className="div-description-amenities-right">
            {this.amenities[1].map((amenity, idx) => <AmenityItem key={idx} amenity={amenity}></AmenityItem>)}
          </div>
        </div>
        <div>
          <a className="a-description-fold" onClick={this.handleOpenModal}>Show all {this.props.amenities.length} amenities</a>
            <ReactModal
             isOpen={this.state.showModal}
             className="modal-description-amenities-container"
             contentLabel="Amenities"
             onRequestClose={this.handleCloseModal}
            >
              <div className="modal-description-amenities-top">
                <div className="div-description-scroll-wrapper">
                  <button className="button-description-modal-close" onClick={this.handleCloseModal}>
                    <span dangerouslySetInnerHTML={{ __html: SVGIcons['Close Button'] }}></span>
                  </button>
                  <ModalAmenities amenities={this.props.amenities}></ModalAmenities>
                </div>
              </div>
            </ReactModal>
        </div>
      </div>
    );
  }
}

export default Amenities;
