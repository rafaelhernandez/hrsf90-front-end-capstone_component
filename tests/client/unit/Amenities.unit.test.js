import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Amenities from '../../../client/src/js/Amenities.jsx';
import descriptionSample from '../../../db/data/description_sample.js';

configure({ adapter: new Adapter() });

describe('Amenities', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Amenities amenities={descriptionSample.amenities}></Amenities>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
