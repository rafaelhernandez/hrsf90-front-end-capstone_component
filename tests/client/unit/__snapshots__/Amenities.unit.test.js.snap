// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amenities should render correctly 1`] = `
<div
  className="div-description-amenities"
>
  <AmenityItem
    amenity="TV"
    key="0"
  />
  <AmenityItem
    amenity="Cable TV"
    key="1"
  />
  <AmenityItem
    amenity="Wifi"
    key="2"
  />
  <AmenityItem
    amenity="Pets live on this property"
    key="3"
  />
  <AmenityItem
    amenity="Indoor fireplace"
    key="4"
  />
  <AmenityItem
    amenity="Heating"
    key="5"
  />
  <AmenityItem
    amenity="Family/kid friendly"
    key="6"
  />
  <AmenityItem
    amenity="Washer"
    key="7"
  />
  <AmenityItem
    amenity="Dryer"
    key="8"
  />
  <AmenityItem
    amenity="Smoke detector"
    key="9"
  />
  <AmenityItem
    amenity="Carbon monoxide detector"
    key="10"
  />
  <AmenityItem
    amenity="Essentials"
    key="11"
  />
  <AmenityItem
    amenity="Hangers"
    key="12"
  />
  <AmenityItem
    amenity="Hair dryer"
    key="13"
  />
  <AmenityItem
    amenity="Iron"
    key="14"
  />
  <AmenityItem
    amenity="Laptop friendly workspace"
    key="15"
  />
  <AmenityItem
    amenity="Private living room"
    key="16"
  />
  <AmenityItem
    amenity="Private entrance"
    key="17"
  />
  <AmenityItem
    amenity="Bathtub"
    key="18"
  />
  <AmenityItem
    amenity="Hot water"
    key="19"
  />
  <AmenityItem
    amenity="Coffee maker"
    key="20"
  />
  <AmenityItem
    amenity="Refrigerator"
    key="21"
  />
  <AmenityItem
    amenity="Dishes and silverware"
    key="22"
  />
  <AmenityItem
    amenity="Oven"
    key="23"
  />
  <AmenityItem
    amenity="Garden or backyard"
    key="24"
  />
  <AmenityItem
    amenity="Hot water kettle"
    key="25"
  />
</div>
`;
